modules/features2d/include/opencv2/features2d.hpp

```diff
@@ -271,10 +271,13 @@ class CV_EXPORTS_W SIFT : public Feature2D
 
     @param sigma The sigma of the Gaussian applied to the input image at the octave \#0. If your image
     is captured with a weak camera with soft lenses, you might want to reduce the number.
+
+    @param useUcharDescriptors If true, the type of descriptors will be CV_8U. If false (default), it
+    will be CV_32F.
     */
     CV_WRAP static Ptr<SIFT> create(int nfeatures = 0, int nOctaveLayers = 3,
         double contrastThreshold = 0.04, double edgeThreshold = 10,
-        double sigma = 1.6);
+        double sigma = 1.6, bool useUcharDescriptors = false);
```

Comment:
I suggest to change the last parameter to `int descriptorType = CV_32F`, because it's more extendible and easier to read the code:

```SIFT::create(..., CV_8U);```

is more clear than

```SIFT::create(..., true);```